flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/allocator/SlotAssignerTest.java

```diff
@@ -0,0 +1,170 @@
+            Arrays.asList(
+                    slot1OfTml1,
+                    slot2OfTml1,
+                    slot3OfTml1,
+                    slot1OfTml2,
+                    slot2OfTml2,
+                    slot3OfTml2,
+                    slot1OfTml3,
+                    slot2OfTml3);
+
+    private static Stream<Arguments> getTestingParameters() {
+        return Stream.of(
+                Arguments.of(
+                        new StateLocalitySlotAssigner(),
+                        3,
+                        allSlots,
+                        createTestingScores(Tuple2.of(slot1OfTml1, 2L), Tuple2.of(slot1OfTml2, 1L)),
+                        Arrays.asList(tml1, tml3)),
+                Arguments.of(
+                        new StateLocalitySlotAssigner(),
+                        2,
+                        allSlots,
+                        createTestingScores(Tuple2.of(slot1OfTml1, 2L), Tuple2.of(slot1OfTml2, 2L)),
+                        Collections.singletonList(tml3)),
+                Arguments.of(
+                        new StateLocalitySlotAssigner(),
+                        6,
+                        allSlots,
+                        createTestingScores(Tuple2.of(slot1OfTml2, 2L)),
+                        Arrays.asList(tml1, tml2, tml3)),
+                Arguments.of(
+                        new StateLocalitySlotAssigner(),
+                        4,
+                        Arrays.asList(
+                                slot1OfTml1,
+                                slot2OfTml1,
+                                slot1OfTml2,
+                                slot2OfTml2,
+                                slot1OfTml3,
+                                slot2OfTml3),
+                        createTestingScores(Tuple2.of(slot1OfTml2, 2L), Tuple2.of(slot2OfTml3, 1L)),
+                        Arrays.asList(tml2, tml3)),
+                Arguments.of(
+                        new DefaultSlotAssigner(),
+                        2,
+                        allSlots,
+                        null,
+                        Collections.singletonList(tml3)),
+                Arguments.of(
+                        new DefaultSlotAssigner(),
+                        3,
+                        Arrays.asList(slot1OfTml1, slot1OfTml2, slot2OfTml2, slot3OfTml2),
+                        null,
+                        Arrays.asList(tml1, tml2)),
+                Arguments.of(
+                        new DefaultSlotAssigner(),
+                        7,
+                        allSlots,
+                        createTestingScores(Tuple2.of(slot1OfTml2, 2L)),
+                        Arrays.asList(tml1, tml2, tml3)));
+    }
+
+    @MethodSource("getTestingParameters")
+    @ParameterizedTest(
+            name =
+                    "slotAssigner={0}, groups={1}, allSlots={2}, scoredAllocations={3}, minimalTaskExecutors={4}")
+    void testPickSlotsInMinimalTaskExecutors(
```

Comment:
I’m think could we call `assignSlots` instead of `pickSlotsInMinimalTaskExecutors` to test `PickSlotsInMinimalTaskExecutors`?

It's for 3 reasons:
1. We don't need to care about the inside logic of SlotAssigner.
2. Current test cannot cover the following case: 
    - If `getSortedTaskExecutors` and `pickSlotsInMinimalTaskExecutors` doesn't have bug, but `SlotAssigner.assignSlots` doesn't call them correctly. 
3. Current test doesn't need to check `slotAssigner instanceof`, it could call `slotAssigner.assignSlots` directly.